app/components/createContainer/createContainerController.js

```diff
@@ -232,7 +233,174 @@ function ($q, $scope, $state, $stateParams, $filter, Container, ContainerHelper,
     return config;
   }
 
+  function confirmCreateContainer(cb) {
+    Container.query({
+      all: 1,
+      filters: {name: [$scope.config.name]}
+    }, function (data) {
+      var confirmDialog = false;
+      var containerId;
+      // Prompt if we found name to confirm replacement
+      for (var c in data) {
+        for (var n in data[c].Names) {
+          if (data[c].Names[n] === '/' + $scope.config.name) {
+            confirmDialog = true;
+            containerId = data[c].Id;
+          }
+        }
+      }
+      if (confirmDialog) {
+        ModalService.confirmDeletion(
+          'A container with the same name is already present on this host. Do you want to remove it?',
+          function onConfirm(confirmed) {
+            if(!confirmed) { cb(false); }
+            else {
+              // Remove old container
+              Container.remove({id: containerId, v: 0, force: true}, function(d) {
+                if (d.message) {
+                  Notifications.error("Error", d, "Unable to remove container");
+                  cb(false);
+                } else {
+                  if (c.Metadata && c.Metadata.ResourceControl) {
+                    ResourceControlService.removeContainerResourceControl(c.Metadata.ResourceControl.OwnerId, containerId)
+                    .then(function success() {
+                      Notifications.success("Container Removed", containerId);
+                      cb(true);
+                    })
+                    .catch(function error(err) {
+                      Notifications.error("Failure", err, "Unable to remove container ownership");
+                      cb(false);
+                    });
+                  } else {
+                    Notifications.success("Container Removed", containerId);
+                    cb(true);
+                  }
+                }
+              });
+            }
+          }
+        );
+      } else {
+        cb(true);
+      }
+    }, function error(err) {
+      cb(false);
+      Notifications.error("Failure", err, "Unable to retrieve containers");
+    });
+  }
+
+  function loadFromContainerSpec() {
+    // Get container
+    Container.get({id: $stateParams.from}, function(d) {
+      // Add Config
+      $scope.config = d.Config;
```

Comment:
Again, all the code related to creating a new config object from an existing container should be externalized in `ContainerHelper`.